plugins/hydra_optuna_sweeper/tests/test_optuna_sweeper_plugin.py

```diff
@@ -0,0 +1,129 @@
+
+import pytest
+from hydra.core.override_parser.overrides_parser import OverridesParser
+from hydra.core.plugins import Plugins
+from hydra.plugins.sweeper import Sweeper
+from hydra.test_utils.test_utils import TSweepRunner, chdir_plugin_root, get_run_output
+from omegaconf import DictConfig, OmegaConf
+from optuna.distributions import (
+    BaseDistribution,
+    CategoricalDistribution,
+    DiscreteUniformDistribution,
+    IntLogUniformDistribution,
+    IntUniformDistribution,
+    LogUniformDistribution,
+    UniformDistribution,
+)
+
+from hydra_plugins.hydra_optuna_sweeper import _impl
+from hydra_plugins.hydra_optuna_sweeper.optuna_sweeper import OptunaSweeper
+
+chdir_plugin_root()
+
+
+def test_discovery() -> None:
+    assert OptunaSweeper.__name__ in [
+        x.__name__ for x in Plugins.instance().discover(Sweeper)
+    ]
+
+
+def check_distribution(expected: BaseDistribution, actual: BaseDistribution) -> None:
+    if not isinstance(expected, CategoricalDistribution):
+        assert expected == actual
+        return
+
+    assert isinstance(actual, CategoricalDistribution)
+    # shuffle() will randomize the order of items in choices.
+    assert set(expected.choices) == set(actual.choices)
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "input, expected",
+    [
+        (
+            {"type": "categorical", "choices": [1, 2, 3]},
+            CategoricalDistribution([1, 2, 3]),
+        ),
+        ({"type": "int", "low": 0, "high": 10}, IntUniformDistribution(0, 10)),
+        (
+            {"type": "int", "low": 0, "high": 10, "step": 2},
+            IntUniformDistribution(0, 10, step=2),
+        ),
+        ({"type": "int", "low": 0, "high": 5}, IntUniformDistribution(0, 5)),
+        (
+            {"type": "int", "low": 1, "high": 100, "log": True},
+            IntLogUniformDistribution(1, 100),
+        ),
+        ({"type": "float", "low": 0, "high": 1}, UniformDistribution(0, 1)),
+        (
+            {"type": "float", "low": 0, "high": 10, "step": 2},
+            DiscreteUniformDistribution(0, 10, 2),
+        ),
+        (
+            {"type": "float", "low": 1, "high": 100, "log": True},
+            LogUniformDistribution(1, 100),
+        ),
+    ],
+)
+def test_create_optuna_distribution_from_config(input: Any, expected: Any) -> None:
+    actual = _impl.create_optuna_distribution_from_config(input)
+    check_distribution(expected, actual)
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "input, expected",
+    [
+        ("key=choice(1,2)", CategoricalDistribution([1, 2])),
+        ("key=choice('hello', 'world')", CategoricalDistribution(["hello", "world"])),
+        ("key=shuffle(range(1,3))", CategoricalDistribution((1, 2))),
+        ("key=range(1,3)", IntUniformDistribution(1, 3)),
+        ("key=interval(1, 5)", UniformDistribution(1, 5)),
+        ("key=tag(int, interval(1, 5))", IntUniformDistribution(1, 5)),
+        ("key=tag(log, interval(1, 5))", LogUniformDistribution(1, 5)),
```

Comment:
You are missing a test for LogInt.

Reply:
You're right. I added it in https://github.com/facebookresearch/hydra/pull/1132/commits/22c6e4fd9b4e0236dfaf5f0ee3dd7afcc9a21cbd.